mythtv/programs/scripts/metadata/Movie/tmdb3.py

```diff
@@ -358,22 +547,42 @@ def main():
     if opts.country:
         set_locale(country=opts.country, fallthrough=True)
 
-    if (len(args) != 1) or (args[0] == ''):
-        sys.stdout.write('ERROR: tmdb3.py requires exactly one non-empty argument')
+    if (len(args) < 1) or (args[0] == ''):
+        sys.stdout.write('ERROR: tmdb3.py requires at least one non-empty argument.\n')
+        sys.exit(1)
+
+    if opts.moviedata and opts.showType == 'television' and len(args) < 3:
+        sys.stdout.write('ERROR: tmdb3.py -D requires three non-empty arguments for television details.\n')
         sys.exit(1)
 
     try:
         if opts.movielist:
-            buildList(args[0], opts)
+            if opts.showType == 'movie':
+                buildMovieList(args[0], opts)
+            else:
+                buildTVList(args[0], opts)
+
+        if opts.numbers and opts.showType == 'television':
+            buildEpisode(args[0:2], opts)
 
         if opts.moviedata:
-            buildSingle(args[0], opts)
+            if opts.showType == 'movie':
+                buildSingle(args[0], opts)
+            else:
+                buildEpisode(args[0:3], opts)
 
         if opts.collectiondata:
-            buildCollection(args[0], opts)
+            if opts.showType == 'television':
+                buildCollection(args[0], opts)
+            else:
+                sys.stdout.write('ERROR: tmdb3.py -C only applies if -T television is selected.\n')
+                sys.exit(1)
+
     except RuntimeError as exc:
         sys.stdout.write('ERROR: ' + str(exc) + ' exception')
+        import traceback
+        traceback.print_exc()
         sys.exit(1)
 
 if __name__ == '__main__':
-    main()
+    main("movie",'tmdb3.py')
```

Comment:
A good construct is sys.exit(main(....)) and use return in other functions to allow for easier unit tests.